The job **`aarch64-gnu-llvm-19-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/15362649803/job/43231716075) [(plain)](https://github.com/rust-lang/rust/commit/8180ed054365710131a9718e21fe3f35854b5c6c/checks/43231716075/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] build_script_build test:false 0.297
error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
   --> library/core/src/num/nonzero.rs:433:52
    |
433 |     pub unsafe fn from_mut_unchecked(n: &mut T) -> &mut Self {
    |                                                    ^^^^^^^^^
    |
    = note: lifetimes appearing in an associated or opaque type are not considered constrained
    = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
  --> library/core/src/ops/index.rs:72:36
   |
72 |     fn index(&self, index: Idx) -> &Self::Output;
   |                                    ^^^^^^^^^^^^^
   |
   = note: lifetimes appearing in an associated or opaque type are not considered constrained
   = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
   --> library/core/src/ops/index.rs:177:44
    |
177 |     fn index_mut(&mut self, index: Idx) -> &mut Self::Output;
    |                                            ^^^^^^^^^^^^^^^^^
    |
    = note: lifetimes appearing in an associated or opaque type are not considered constrained
    = note: consider introducing a named lifetime parameter

---

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
   --> library/core/src/slice/mod.rs:635:56
    |
635 |     pub unsafe fn get_unchecked<I>(&self, index: I) -> &I::Output
    |                                                        ^^^^^^^^^^
    |
    = note: lifetimes appearing in an associated or opaque type are not considered constrained
    = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
   --> library/core/src/slice/mod.rs:679:64
    |
679 |     pub unsafe fn get_unchecked_mut<I>(&mut self, index: I) -> &mut I::Output
    |                                                                ^^^^^^^^^^^^^^
    |
    = note: lifetimes appearing in an associated or opaque type are not considered constrained
    = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1107:49
     |
1107 |     pub const fn windows(&self, size: usize) -> Windows<'_, T> {
     |                                                 ^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

---

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1314:71
     |
1314 |     pub const unsafe fn as_chunks_unchecked<const N: usize>(&self) -> &[[T; N]] {
     |                                                                       ^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1372:54
     |
1372 |     pub const fn as_chunks<const N: usize>(&self) -> (&[[T; N]], &[T]) {
     |                                                      ^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1419:55
     |
1419 |     pub const fn as_rchunks<const N: usize>(&self) -> (&[T], &[[T; N]]) {
     |                                                       ^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1460:57
     |
1460 |     pub const fn array_chunks<const N: usize>(&self) -> ArrayChunks<'_, T, N> {
     |                                                         ^^^^^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1510:79
     |
1510 |     pub const unsafe fn as_chunks_unchecked_mut<const N: usize>(&mut self) -> &mut [[T; N]] {
     |                                                                               ^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1564:62
     |
1564 |     pub const fn as_chunks_mut<const N: usize>(&mut self) -> (&mut [[T; N]], &mut [T]) {
     |                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1617:63
     |
1617 |     pub const fn as_rchunks_mut<const N: usize>(&mut self) -> (&mut [T], &mut [[T; N]]) {
     |                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1660:65
     |
1660 |     pub const fn array_chunks_mut<const N: usize>(&mut self) -> ArrayChunksMut<'_, T, N> {
     |                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1694:58
     |
1694 |     pub const fn array_windows<const N: usize>(&self) -> ArrayWindows<'_, T, N> {
     |                                                          ^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1730:55
     |
1730 |     pub const fn rchunks(&self, chunk_size: usize) -> RChunks<'_, T> {
     |                                                       ^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1770:63
     |
1770 |     pub const fn rchunks_mut(&mut self, chunk_size: usize) -> RChunksMut<'_, T> {
     |                                                               ^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1811:61
     |
1811 |     pub const fn rchunks_exact(&self, chunk_size: usize) -> RChunksExact<'_, T> {
     |                                                             ^^^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1856:69
     |
1856 |     pub const fn rchunks_exact_mut(&mut self, chunk_size: usize) -> RChunksExactMut<'_, T> {
     |                                                                     ^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:1984:49
     |
1984 |     pub const fn split_at(&self, mid: usize) -> (&[T], &[T]) {
     |                                                 ^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:2018:57
     |
2018 |     pub const fn split_at_mut(&mut self, mid: usize) -> (&mut [T], &mut [T]) {
     |                                                         ^^^^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:2070:66
     |
2070 |     pub const unsafe fn split_at_unchecked(&self, mid: usize) -> (&[T], &[T]) {
     |                                                                  ^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:2124:74
     |
2124 |     pub const unsafe fn split_at_mut_unchecked(&mut self, mid: usize) -> (&mut [T], &mut [T]) {
     |                                                                          ^^^^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error[E0581]: return type references an anonymous lifetime, which is not constrained by the fn input types
    --> library/core/src/slice/mod.rs:4656:10
     |
4656 |     ) -> [&mut I::Output; N]
     |          ^^^^^^^^^^^^^^^^^^^
     |
     = note: lifetimes appearing in an associated or opaque type are not considered constrained
     = note: consider introducing a named lifetime parameter

error: internal compiler error: compiler/rustc_middle/src/ty/context.rs:2839:17: args not compatible with generics for function::FnMut: args=[
                                    (A/#1,),
                                ], generics=[
                                    Self/#0,
                                    Args/#1,
                                ]


thread 'rustc' panicked at compiler/rustc_middle/src/ty/context.rs:2839:17:
Box<dyn Any>
---
   4:     0xff9cd2e36b6c - std::panicking::default_hook::{{closure}}::h068c040e92fd820f
   5:     0xff9cd2e36970 - std::panicking::default_hook::ha6a4fa006862e418
   6:     0xff9ccef83354 - <alloc[38258f7e2bf206cd]::boxed::Box<rustc_driver_impl[fc38ece42c64b183]::install_ice_hook::{closure#1}> as core[f528e8398041c3]::ops::function::Fn<(&dyn for<'a, 'b> core[f528e8398041c3]::ops::function::Fn<(&'a std[106e72d616e4ce2]::panic::PanicHookInfo<'b>,), Output = ()> + core[f528e8398041c3]::marker::Sync + core[f528e8398041c3]::marker::Send, &std[106e72d616e4ce2]::panic::PanicHookInfo)>>::call
   7:     0xff9cd2e3776c - std::panicking::rust_panic_with_hook::h675f4f76233f4a65
   8:     0xff9cd29bc184 - std[106e72d616e4ce2]::panicking::begin_panic::<rustc_errors[5a64fae8a08da1b3]::ExplicitBug>::{closure#0}
   9:     0xff9cd29b5060 - std[106e72d616e4ce2]::sys::backtrace::__rust_end_short_backtrace::<std[106e72d616e4ce2]::panicking::begin_panic<rustc_errors[5a64fae8a08da1b3]::ExplicitBug>::{closure#0}, !>
  10:     0xff9ccee2952c - std[106e72d616e4ce2]::panicking::begin_panic::<rustc_errors[5a64fae8a08da1b3]::ExplicitBug>
  11:     0xff9ccee29800 - <rustc_errors[5a64fae8a08da1b3]::diagnostic::BugAbort as rustc_errors[5a64fae8a08da1b3]::diagnostic::EmissionGuarantee>::emit_producing_guarantee
  12:     0xff9cd27f5110 - rustc_middle[38c7c9547c2dc403]::util::bug::opt_span_bug_fmt::<rustc_span[228dcb02b9b5061]::span_encoding::Span>::{closure#0}
  13:     0xff9cd27f4478 - rustc_middle[38c7c9547c2dc403]::ty::context::tls::with_opt::<rustc_middle[38c7c9547c2dc403]::util::bug::opt_span_bug_fmt<rustc_span[228dcb02b9b5061]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  14:     0xff9cd27f4438 - rustc_middle[38c7c9547c2dc403]::ty::context::tls::with_context_opt::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::with_opt<rustc_middle[38c7c9547c2dc403]::util::bug::opt_span_bug_fmt<rustc_span[228dcb02b9b5061]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  15:     0xff9ccee1a4a8 - rustc_middle[38c7c9547c2dc403]::util::bug::bug_fmt
  16:     0xff9cd263f4ac - <rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt>::debug_assert_args_compatible
  17:     0xff9ccf946710 - <rustc_type_ir[ce91a61ad37af357]::predicate::TraitRef<rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt>>::with_self_ty
  18:     0xff9ccf8e7394 - <rustc_hir_typeck[2d43da27e89bd8e3]::typeck_root_ctxt::TypeckRootCtxt>::register_predicate
  19:     0xff9ccf8e2190 - <rustc_hir_typeck[2d43da27e89bd8e3]::typeck_root_ctxt::TypeckRootCtxt>::register_predicates::<thin_vec[3115d7c89c0cf5b8]::ThinVec<rustc_infer[2aaf44c447e2c216]::traits::Obligation<rustc_middle[38c7c9547c2dc403]::ty::predicate::Predicate>>>
  20:     0xff9ccf84655c - rustc_hir_typeck[2d43da27e89bd8e3]::check::check_fn
  21:     0xff9ccf8516a8 - rustc_hir_typeck[2d43da27e89bd8e3]::typeck_with_inspect
  22:     0xff9cd1592cb4 - rustc_query_impl[fd050f0a0d409e3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[fd050f0a0d409e3a]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  23:     0xff9cd14da00c - <rustc_query_impl[fd050f0a0d409e3a]::query_impl::typeck::dynamic_query::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt, rustc_span[228dcb02b9b5061]::def_id::LocalDefId)>>::call_once
  24:     0xff9cd1521ae8 - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<rustc_query_system[dee336ee533e7cf2]::query::plumbing::execute_job_non_incr<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  25:     0xff9cd144d63c - rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_execute_query::<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt, false>
  26:     0xff9cd175ba7c - rustc_query_impl[fd050f0a0d409e3a]::query_impl::typeck::get_query_non_incr::__rust_end_short_backtrace
  27:     0xff9cd019e610 - rustc_mir_build[f21dffcc6e88824]::thir::pattern::check_match::check_match
  28:     0xff9cd15748b4 - rustc_query_impl[fd050f0a0d409e3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[fd050f0a0d409e3a]::query_impl::check_match::dynamic_query::{closure#2}::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>>
  29:     0xff9cd1496868 - <rustc_query_impl[fd050f0a0d409e3a]::query_impl::check_match::dynamic_query::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt, rustc_span[228dcb02b9b5061]::def_id::LocalDefId)>>::call_once
  30:     0xff9cd1521a80 - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<rustc_query_system[dee336ee533e7cf2]::query::plumbing::execute_job_non_incr<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, true, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>>
  31:     0xff9cd1446f1c - rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_execute_query::<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt, false>
  32:     0xff9cd172f108 - rustc_query_impl[fd050f0a0d409e3a]::query_impl::check_match::get_query_non_incr::__rust_end_short_backtrace
  33:     0xff9cd011feb4 - rustc_middle[38c7c9547c2dc403]::query::plumbing::query_ensure_error_guaranteed::<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, ()>
  34:     0xff9cd0123dd8 - rustc_mir_build[f21dffcc6e88824]::builder::build_mir
  35:     0xff9ccfefcad4 - rustc_mir_transform[c79836063c7fcf87]::mir_built
  36:     0xff9cd1594428 - rustc_query_impl[fd050f0a0d409e3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[fd050f0a0d409e3a]::query_impl::mir_built::dynamic_query::{closure#2}::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  37:     0xff9cd14dd4f8 - <rustc_query_impl[fd050f0a0d409e3a]::query_impl::mir_built::dynamic_query::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt, rustc_span[228dcb02b9b5061]::def_id::LocalDefId)>>::call_once
  38:     0xff9cd1521ae8 - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<rustc_query_system[dee336ee533e7cf2]::query::plumbing::execute_job_non_incr<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  39:     0xff9cd144d63c - rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_execute_query::<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt, false>
  40:     0xff9cd17596ac - rustc_query_impl[fd050f0a0d409e3a]::query_impl::mir_built::get_query_non_incr::__rust_end_short_backtrace
  41:     0xff9ccfebf49c - rustc_mir_transform[c79836063c7fcf87]::ffi_unwind_calls::has_ffi_unwind_calls
  42:     0xff9cd1585018 - rustc_query_impl[fd050f0a0d409e3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[fd050f0a0d409e3a]::query_impl::has_ffi_unwind_calls::dynamic_query::{closure#2}::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>>
  43:     0xff9cd14bb53c - <rustc_query_impl[fd050f0a0d409e3a]::query_impl::has_ffi_unwind_calls::dynamic_query::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt, rustc_span[228dcb02b9b5061]::def_id::LocalDefId)>>::call_once
  44:     0xff9cd1521a80 - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<rustc_query_system[dee336ee533e7cf2]::query::plumbing::execute_job_non_incr<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, true, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>>
  45:     0xff9cd1446f1c - rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_execute_query::<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt, false>
  46:     0xff9cd17332cc - rustc_query_impl[fd050f0a0d409e3a]::query_impl::has_ffi_unwind_calls::get_query_non_incr::__rust_end_short_backtrace
  47:     0xff9ccfefcf34 - rustc_mir_transform[c79836063c7fcf87]::mir_promoted
  48:     0xff9cd1576b88 - rustc_query_impl[fd050f0a0d409e3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[fd050f0a0d409e3a]::query_impl::mir_promoted::dynamic_query::{closure#2}::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 16usize]>>
  49:     0xff9cd149b9f8 - <rustc_query_impl[fd050f0a0d409e3a]::query_impl::mir_promoted::dynamic_query::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt, rustc_span[228dcb02b9b5061]::def_id::LocalDefId)>>::call_once
  50:     0xff9cd1522224 - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_load_from_disk_and_cache_in_memory<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::CrateNum, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 16usize]>>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 16usize]>>
  51:     0xff9cd1443bd0 - rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_execute_query::<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 16usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt, false>
  52:     0xff9cd17ea254 - rustc_query_impl[fd050f0a0d409e3a]::query_impl::mir_promoted::get_query_non_incr::__rust_end_short_backtrace
  53:     0xff9cd03abb28 - rustc_borrowck[eabd22c009dddb67]::mir_borrowck
  54:     0xff9cd1576840 - rustc_query_impl[fd050f0a0d409e3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[fd050f0a0d409e3a]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  55:     0xff9cd149b240 - <rustc_query_impl[fd050f0a0d409e3a]::query_impl::mir_borrowck::dynamic_query::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt, rustc_span[228dcb02b9b5061]::def_id::LocalDefId)>>::call_once
  56:     0xff9cd1521ae8 - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<rustc_query_system[dee336ee533e7cf2]::query::plumbing::execute_job_non_incr<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  57:     0xff9cd144d63c - rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_execute_query::<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_data_structures[f70675bad9a4d67]::vec_cache::VecCache<rustc_span[228dcb02b9b5061]::def_id::LocalDefId, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[dee336ee533e7cf2]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt, false>
  58:     0xff9cd175bc24 - rustc_query_impl[fd050f0a0d409e3a]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  59:     0xff9ccfc49c50 - rustc_hir_analysis[dfe4b29c205cfad8]::collect::type_of::opaque::find_opaque_ty_constraints_for_rpit
  60:     0xff9ccfc4b7e0 - rustc_hir_analysis[dfe4b29c205cfad8]::collect::type_of::type_of_opaque
  61:     0xff9cd1579e84 - rustc_query_impl[fd050f0a0d409e3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[fd050f0a0d409e3a]::query_impl::type_of_opaque::dynamic_query::{closure#2}::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  62:     0xff9cd14a2ba0 - <rustc_query_impl[fd050f0a0d409e3a]::query_impl::type_of_opaque::dynamic_query::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt, rustc_span[228dcb02b9b5061]::def_id::DefId)>>::call_once
  63:     0xff9cd15207dc - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_load_from_disk_and_cache_in_memory<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_query_system[dee336ee533e7cf2]::query::caches::DefIdCache<rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  64:     0xff9cd13d860c - rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_execute_query::<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_query_system[dee336ee533e7cf2]::query::caches::DefIdCache<rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt, false>
  65:     0xff9cd165bb20 - rustc_query_impl[fd050f0a0d409e3a]::query_impl::type_of_opaque::get_query_non_incr::__rust_end_short_backtrace
  66:     0xff9ccfc4a82c - rustc_hir_analysis[dfe4b29c205cfad8]::collect::type_of::type_of
  67:     0xff9cd1593004 - rustc_query_impl[fd050f0a0d409e3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[fd050f0a0d409e3a]::query_impl::type_of::dynamic_query::{closure#2}::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  68:     0xff9cd14da86c - <rustc_query_impl[fd050f0a0d409e3a]::query_impl::type_of::dynamic_query::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt, rustc_span[228dcb02b9b5061]::def_id::DefId)>>::call_once
  69:     0xff9cd15207dc - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_load_from_disk_and_cache_in_memory<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_query_system[dee336ee533e7cf2]::query::caches::DefIdCache<rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  70:     0xff9cd13d860c - rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_execute_query::<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_query_system[dee336ee533e7cf2]::query::caches::DefIdCache<rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt, false>
  71:     0xff9cd16c1ea4 - rustc_query_impl[fd050f0a0d409e3a]::query_impl::type_of::get_query_non_incr::__rust_end_short_backtrace
  72:     0xff9cd10102d0 - <rustc_privacy[ff420c349c7ec657]::ReachEverythingInTheInterfaceVisitor>::ty
  73:     0xff9cd1015078 - rustc_privacy[ff420c349c7ec657]::effective_visibilities
  74:     0xff9cd15872dc - rustc_query_impl[fd050f0a0d409e3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[fd050f0a0d409e3a]::query_impl::effective_visibilities::dynamic_query::{closure#2}::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  75:     0xff9cd14c0448 - <rustc_query_impl[fd050f0a0d409e3a]::query_impl::effective_visibilities::dynamic_query::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt, ())>>::call_once
  76:     0xff9cd1520ba0 - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<rustc_query_system[dee336ee533e7cf2]::query::plumbing::execute_job_non_incr<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_query_system[dee336ee533e7cf2]::query::caches::SingleCache<rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>
  77:     0xff9cd13e440c - rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_execute_query::<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_query_system[dee336ee533e7cf2]::query::caches::SingleCache<rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt, false>
  78:     0xff9cd172f1c4 - rustc_query_impl[fd050f0a0d409e3a]::query_impl::effective_visibilities::get_query_non_incr::__rust_end_short_backtrace
  79:     0xff9cd0db7d34 - rustc_passes[e31e0f19a3f328f4]::stability::check_unused_or_stable_features
  80:     0xff9ccf175968 - <rustc_session[b8fe48274d8e4ffe]::session::Session>::time::<(), rustc_interface[5342e0e39e48f021]::passes::run_required_analyses::{closure#0}::{closure#1}::{closure#0}::{closure#2}::{closure#0}>
  81:     0xff9ccf1d0860 - <rustc_data_structures[f70675bad9a4d67]::sync::parallel::ParallelGuard>::run::<(), rustc_interface[5342e0e39e48f021]::passes::run_required_analyses::{closure#0}::{closure#0}::{closure#2}>
  82:     0xff9ccf18fd64 - rustc_interface[5342e0e39e48f021]::passes::analysis
  83:     0xff9cd1593198 - rustc_query_impl[fd050f0a0d409e3a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[fd050f0a0d409e3a]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 0usize]>>
  84:     0xff9cd14dac24 - <rustc_query_impl[fd050f0a0d409e3a]::query_impl::analysis::dynamic_query::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt, ())>>::call_once
  85:     0xff9cd15208d8 - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<rustc_query_system[dee336ee533e7cf2]::query::plumbing::execute_job_non_incr<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_query_system[dee336ee533e7cf2]::query::caches::SingleCache<rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 0usize]>>::{closure#0}, rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 0usize]>>
  86:     0xff9cd13dba30 - rustc_query_system[dee336ee533e7cf2]::query::plumbing::try_execute_query::<rustc_query_impl[fd050f0a0d409e3a]::DynamicConfig<rustc_query_system[dee336ee533e7cf2]::query::caches::SingleCache<rustc_middle[38c7c9547c2dc403]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[fd050f0a0d409e3a]::plumbing::QueryCtxt, false>
  87:     0xff9cd169d034 - rustc_query_impl[fd050f0a0d409e3a]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  88:     0xff9ccef6ee30 - <std[106e72d616e4ce2]::thread::local::LocalKey<core[f528e8398041c3]::cell::Cell<*const ()>>>::with::<rustc_middle[38c7c9547c2dc403]::ty::context::tls::enter_context<<rustc_middle[38c7c9547c2dc403]::ty::context::GlobalCtxt>::enter<rustc_interface[5342e0e39e48f021]::passes::create_and_enter_global_ctxt<core[f528e8398041c3]::option::Option<rustc_interface[5342e0e39e48f021]::queries::Linker>, rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}, core[f528e8398041c3]::option::Option<rustc_interface[5342e0e39e48f021]::queries::Linker>>::{closure#1}, core[f528e8398041c3]::option::Option<rustc_interface[5342e0e39e48f021]::queries::Linker>>::{closure#0}, core[f528e8398041c3]::option::Option<rustc_interface[5342e0e39e48f021]::queries::Linker>>
  89:     0xff9ccef37df8 - <rustc_middle[38c7c9547c2dc403]::ty::context::TyCtxt>::create_global_ctxt::<core[f528e8398041c3]::option::Option<rustc_interface[5342e0e39e48f021]::queries::Linker>, rustc_interface[5342e0e39e48f021]::passes::create_and_enter_global_ctxt<core[f528e8398041c3]::option::Option<rustc_interface[5342e0e39e48f021]::queries::Linker>, rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}::{closure#2}>::{closure#2}::{closure#0}>
  90:     0xff9ccef10ab4 - <rustc_interface[5342e0e39e48f021]::passes::create_and_enter_global_ctxt<core[f528e8398041c3]::option::Option<rustc_interface[5342e0e39e48f021]::queries::Linker>, rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}::{closure#2}>::{closure#2} as core[f528e8398041c3]::ops::function::FnOnce<(&rustc_session[b8fe48274d8e4ffe]::session::Session, rustc_middle[38c7c9547c2dc403]::ty::context::CurrentGcx, alloc[38258f7e2bf206cd]::sync::Arc<rustc_data_structures[f70675bad9a4d67]::jobserver::Proxy>, &std[106e72d616e4ce2]::sync::once_lock::OnceLock<rustc_middle[38c7c9547c2dc403]::ty::context::GlobalCtxt>, &rustc_data_structures[f70675bad9a4d67]::sync::worker_local::WorkerLocal<rustc_middle[38c7c9547c2dc403]::arena::Arena>, &rustc_data_structures[f70675bad9a4d67]::sync::worker_local::WorkerLocal<rustc_hir[1d8fab2f78d89cca]::Arena>, rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}::{closure#2})>>::call_once::{shim:vtable#0}
  91:     0xff9ccef8290c - <alloc[38258f7e2bf206cd]::boxed::Box<dyn for<'a> core[f528e8398041c3]::ops::function::FnOnce<(&'a rustc_session[b8fe48274d8e4ffe]::session::Session, rustc_middle[38c7c9547c2dc403]::ty::context::CurrentGcx, alloc[38258f7e2bf206cd]::sync::Arc<rustc_data_structures[f70675bad9a4d67]::jobserver::Proxy>, &'a std[106e72d616e4ce2]::sync::once_lock::OnceLock<rustc_middle[38c7c9547c2dc403]::ty::context::GlobalCtxt<'a>>, &'a rustc_data_structures[f70675bad9a4d67]::sync::worker_local::WorkerLocal<rustc_middle[38c7c9547c2dc403]::arena::Arena<'a>>, &'a rustc_data_structures[f70675bad9a4d67]::sync::worker_local::WorkerLocal<rustc_hir[1d8fab2f78d89cca]::Arena<'a>>, rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}::{closure#2}), Output = core[f528e8398041c3]::option::Option<rustc_interface[5342e0e39e48f021]::queries::Linker>>> as core[f528e8398041c3]::ops::function::FnOnce<(&rustc_session[b8fe48274d8e4ffe]::session::Session, rustc_middle[38c7c9547c2dc403]::ty::context::CurrentGcx, alloc[38258f7e2bf206cd]::sync::Arc<rustc_data_structures[f70675bad9a4d67]::jobserver::Proxy>, &std[106e72d616e4ce2]::sync::once_lock::OnceLock<rustc_middle[38c7c9547c2dc403]::ty::context::GlobalCtxt>, &rustc_data_structures[f70675bad9a4d67]::sync::worker_local::WorkerLocal<rustc_middle[38c7c9547c2dc403]::arena::Arena>, &rustc_data_structures[f70675bad9a4d67]::sync::worker_local::WorkerLocal<rustc_hir[1d8fab2f78d89cca]::Arena>, rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}::{closure#2})>>::call_once
  92:     0xff9ccef0f2b4 - rustc_interface[5342e0e39e48f021]::passes::create_and_enter_global_ctxt::<core[f528e8398041c3]::option::Option<rustc_interface[5342e0e39e48f021]::queries::Linker>, rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}::{closure#2}>
  93:     0xff9ccef16460 - <scoped_tls[43361420308e6c2b]::ScopedKey<rustc_span[228dcb02b9b5061]::SessionGlobals>>::set::<rustc_interface[5342e0e39e48f021]::util::run_in_thread_with_globals<rustc_interface[5342e0e39e48f021]::util::run_in_thread_pool_with_globals<rustc_interface[5342e0e39e48f021]::interface::run_compiler<(), rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}, ()>
  94:     0xff9cceebb03c - rustc_span[228dcb02b9b5061]::create_session_globals_then::<(), rustc_interface[5342e0e39e48f021]::util::run_in_thread_with_globals<rustc_interface[5342e0e39e48f021]::util::run_in_thread_pool_with_globals<rustc_interface[5342e0e39e48f021]::interface::run_compiler<(), rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}::{closure#0}>
  95:     0xff9ccef26534 - std[106e72d616e4ce2]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[5342e0e39e48f021]::util::run_in_thread_with_globals<rustc_interface[5342e0e39e48f021]::util::run_in_thread_pool_with_globals<rustc_interface[5342e0e39e48f021]::interface::run_compiler<(), rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  96:     0xff9cceed7768 - <<std[106e72d616e4ce2]::thread::Builder>::spawn_unchecked_<rustc_interface[5342e0e39e48f021]::util::run_in_thread_with_globals<rustc_interface[5342e0e39e48f021]::util::run_in_thread_pool_with_globals<rustc_interface[5342e0e39e48f021]::interface::run_compiler<(), rustc_driver_impl[fc38ece42c64b183]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[f528e8398041c3]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
---
warning: the ICE couldn't be written to `/checkout/rustc-ice-2025-05-31T10_21_28-13352.txt`: Read-only file system (os error 30)

note: rustc 1.89.0-nightly (28b7e2a64 2025-05-31) running on aarch64-unknown-linux-gnu

note: compiler flags: --crate-type lib -C opt-level=3 -C embed-bitcode=no -C codegen-units=1 -C debug-assertions=on -C symbol-mangling-version=legacy -Z randomize-layout -Z unstable-options -Z macro-backtrace -C split-debuginfo=off -C prefer-dynamic -C llvm-args=-import-instr-limit=10 -Z inline-mir -Z inline-mir-preserve-debug -Z mir_strip_debuginfo=locals-in-tiny-functions -C link-args=-Wl,-z,origin -C link-args=-Wl,-rpath,$ORIGIN/../lib -C embed-bitcode=yes -C force-frame-pointers=yes -Z crate-attr=doc(html_root_url="https://doc.rust-lang.org/nightly/") -Z binary-dep-depinfo -Z force-unstable-if-unmarked

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
#0 [typeck] type-checking `ops::try_trait::<impl at library/core/src/ops/try_trait.rs:381:1: 381:29>::wrap_mut_1`
#1 [check_match] match-checking `ops::try_trait::<impl at library/core/src/ops/try_trait.rs:381:1: 381:29>::wrap_mut_1`
... and 8 other queries... use `env RUST_BACKTRACE=1` to see the full query stack
For more information about this error, try `rustc --explain E0581`.
[RUSTC-TIMING] core test:false 4.457
error: could not compile `core` (lib) due to 31 previous errors
warning: build failed, waiting for other jobs to finish...
Build completed unsuccessfully in 0:05:25
```

</details>

--- a/compiler/rustc_type_ir/src/interner.rs
+++ b/compiler/rustc_type_ir/src/interner.rs
@@ -4,7 +4,6 @@ use std::ops::Deref;
 
 use rustc_ast_ir::Movability;
 use rustc_index::bit_set::DenseBitSet;
-use smallvec::SmallVec;
 
 use crate::fold::TypeFoldable;
 use crate::inherent::*;
@@ -382,28 +381,45 @@ impl<T, R> CollectAndApply<T, R> for T {
         F: FnOnce(&[T]) -> R,
     {
         // This code is hot enough that it's worth specializing for the most
-        // common length lists, to avoid the overhead of `SmallVec` creation.
-        // Lengths 0, 1, and 2 typically account for ~95% of cases. If
-        // `size_hint` is incorrect a panic will occur via an `unwrap` or an
-        // `assert`.
-        match iter.size_hint() {
-            (0, Some(0)) => {
-                assert!(iter.next().is_none());
-                f(&[])
-            }
-            (1, Some(1)) => {
-                let t0 = iter.next().unwrap();
-                assert!(iter.next().is_none());
-                f(&[t0])
-            }
-            (2, Some(2)) => {
-                let t0 = iter.next().unwrap();
-                let t1 = iter.next().unwrap();
-                assert!(iter.next().is_none());
-                f(&[t0, t1])
-            }
-            _ => f(&iter.collect::<SmallVec<[_; 8]>>()),
-        }
+        // common length lists, to avoid the overhead of `Vec` creation.
+
+        let Some(t0) = iter.next() else {
+            return f(&[]);
+        };
+
+        let Some(t1) = iter.next() else {
+            return f(&[t0]);
+        };
+
+        let Some(t2) = iter.next() else {
+            return f(&[t0, t1]);
+        };
+
+        let Some(t3) = iter.next() else {
+            return f(&[t0, t1, t2]);
+        };
+
+        let Some(t4) = iter.next() else {
+            return f(&[t0, t1, t2, t3]);
+        };
+
+        let Some(t5) = iter.next() else {
+            return f(&[t0, t1, t2, t3, t4]);
+        };
+
+        let Some(t6) = iter.next() else {
+            return f(&[t0, t1, t2, t3, t4, t5]);
+        };
+
+        let Some(t7) = iter.next() else {
+            return f(&[t0, t1, t2, t3, t4, t5, t6]);
+        };
+
+        let Some(t8) = iter.next() else {
+            return f(&[t0, t1, t2, t3, t4, t5, t6, t7]);
+        };
+
+        f(&[t0, t1, t2, t3, t4, t5, t6, t7, t8].into_iter().chain(iter).collect::<Vec<_>>())
     }
 }
 
@@ -419,29 +435,57 @@ impl<T, R, E> CollectAndApply<T, R> for Result<T, E> {
         F: FnOnce(&[T]) -> R,
     {
         // This code is hot enough that it's worth specializing for the most
-        // common length lists, to avoid the overhead of `SmallVec` creation.
-        // Lengths 0, 1, and 2 typically account for ~95% of cases. If
-        // `size_hint` is incorrect a panic will occur via an `unwrap` or an
-        // `assert`, unless a failure happens first, in which case the result
-        // will be an error anyway.
-        Ok(match iter.size_hint() {
-            (0, Some(0)) => {
-                assert!(iter.next().is_none());
-                f(&[])
-            }
-            (1, Some(1)) => {
-                let t0 = iter.next().unwrap()?;
-                assert!(iter.next().is_none());
-                f(&[t0])
-            }
-            (2, Some(2)) => {
-                let t0 = iter.next().unwrap()?;
-                let t1 = iter.next().unwrap()?;
-                assert!(iter.next().is_none());
-                f(&[t0, t1])
-            }
-            _ => f(&iter.collect::<Result<SmallVec<[_; 8]>, _>>()?),
-        })
+        // common length lists, to avoid the overhead of `Vec` creation.
+
+        let Some(t0) = iter.next() else {
+            return Ok(f(&[]));
+        };
+        let t0 = t0?;
+
+        let Some(t1) = iter.next() else {
+            return Ok(f(&[t0]));
+        };
+        let t1 = t1?;
+
+        let Some(t2) = iter.next() else {
+            return Ok(f(&[t0, t1]));
+        };
+        let t2 = t2?;
+
+        let Some(t3) = iter.next() else {
+            return Ok(f(&[t0, t1, t2]));
+        };
+        let t3 = t3?;
+
+        let Some(t4) = iter.next() else {
+            return Ok(f(&[t0, t1, t2, t3]));
+        };
+        let t4 = t4?;
+
+        let Some(t5) = iter.next() else {
+            return Ok(f(&[t0, t1, t2, t3, t4]));
+        };
+        let t5 = t5?;
+
+        let Some(t6) = iter.next() else {
+            return Ok(f(&[t0, t1, t2, t3, t4, t5]));
+        };
+        let t6 = t6?;
+
+        let Some(t7) = iter.next() else {
+            return Ok(f(&[t0, t1, t2, t3, t4, t5, t6]));
+        };
+        let t7 = t7?;
+
+        let Some(t8) = iter.next() else {
+            return Ok(f(&[t0, t1, t2, t3, t4, t5, t6, t7]));
+        };
+        let t8 = t8?;
+
+        Ok(f(&[Ok(t0), Ok(t1), Ok(t2), Ok(t3), Ok(t4), Ok(t5), Ok(t6), Ok(t7), Ok(t8)]
+            .into_iter()
+            .chain(iter)
+            .collect::<Result<Vec<_>, _>>()?))
     }
 }
 

